sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/GenerateColumnAccessor.scala

```diff
@@ -88,16 +92,21 @@ object GenerateColumnAccessor extends CodeGenerator[Seq[DataType], ColumnarItera
         case array: ArrayType => classOf[ArrayColumnAccessor].getName
         case t: MapType => classOf[MapColumnAccessor].getName
       }
-      ctx.addMutableState(accessorCls, accessorName, s"$accessorName = null;")
-
-      val createCode = dt match {
-        case t if ctx.isPrimitiveType(dt) =>
-          s"$accessorName = new $accessorCls(ByteBuffer.wrap(buffers[$index]).order(nativeOrder));"
-        case NullType | StringType | BinaryType =>
-          s"$accessorName = new $accessorCls(ByteBuffer.wrap(buffers[$index]).order(nativeOrder));"
-        case other =>
-          s"""$accessorName = new $accessorCls(ByteBuffer.wrap(buffers[$index]).order(nativeOrder),
-             (${dt.getClass.getName}) columnTypes[$index]);"""
+      ctx.addMutableState(accessorCls, accessorName, "")
+
+      val createCode = {
+        val shortCls = accessorCls.substring(accessorCls.lastIndexOf(".") + 1)
+        dt match {
+          case t if ctx.isPrimitiveType(dt) =>
+            s"$accessorName = get${accessorClasses.getOrElseUpdate(accessorCls, shortCls)}($index);"
```

Comment:
For example, a generated method `getIntColumnAccessor()` still calls ```ColumnAccessor.apply()``.

Do you want to directly call  ```ColumnAccessor.apply()`` from a method ```hasNext()``` instead of calling it thru `getIntColumnAccessor`?